### ISSUE TITLE
lru_cache function calls parameters are not checked

### ISSUE THREAD

**Issue by andrewjw** [python/mypy#16261]
**Bug Report**

The parameters used in a call to a function which is wrapped in a `functools.lru_cache` are not checked.

**To Reproduce**

```python
from datetime import date
from functools import lru_cache

@lru_cache
def f(v: str, at: date) -> str:
    return v

f()
f("abc")
f("abc", date.today())
f("abc", now=date.today())
f("abc", at=date.today())
f("abc", at="wrong_type")
```

**Expected Behavior**

This should raise several errors about the arguments being wrong. Commenting out the line `@lru_cache` correctly generates the following errors.

```
lru_cache_type_test.py:8: error: Missing positional arguments "v", "at" in call to "f"  [call-arg]
lru_cache_type_test.py:9: error: Missing positional argument "at" in call to "f"  [call-arg]
lru_cache_type_test.py:11: error: Unexpected keyword argument "now" for "f"  [call-arg]
lru_cache_type_test.py:13: error: Argument "at" to "f" has incompatible type "str"; expected "date"  [arg-type]
```

**Actual Behavior**

`Success: no issues found in 1 source file`

**Your Environment**

mypy 1.6.0 running on Python 3.11.6

**Comment by Hnasar** [comment]
We encounter this fairly often too. Just linking to some past discussion: https://github.com/python/mypy/issues/5107#issuecomment-1355954910

**Comment by andrewjw** [comment]
Ah, thanks for that. I assumed I wouldn't be the first to see this, but couldn't find this specific problem. I guess I didn't read that thread closely enough. I guess this can be closed unless the situation has changed since then, and it is possible to fix.

**Comment by machin0r** [comment]
I have a PR (#19432) that implements plugin support to fix this issue.

Would appreciate guidance on getting this reviewed. Thanks!

**Comment by cj81499** [comment]
Originally posted https://github.com/python/mypy/issues/19291#issuecomment-3729601276 before realizing it's a dupe of this issue.

> Ran into this today. An even simpler example of how this can cause issues. https://mypy-play.net/?mypy=latest&python=3.14&flags=strict&gist=d555423c3f7f92b172480b0eefbf2de4
